cli/fathom_web/extract.py

```diff
@@ -0,0 +1,144 @@
+import base64
+import mimetypes
+import pathlib
+import shutil
+import re
+
+from click import argument, command, option, Path
+
+
+BASE64_DATA_PATTERN = re.compile(r'(data:(?P<mime>[a-zA-Z0-9]+?/[a-zA-Z0-9\-.+]+?);(\s?charset=utf-8;)?base64,(?P<string>[a-zA-Z0-9+/=]+))')
+BASE_TAG_PATTERN = re.compile(r'<base.*?>')
+OLD_CSP = re.compile(r"default-src 'none'; img-src data:; media-src data:; style-src data: 'unsafe-inline'; font-src data:; frame-src data:")
+NEW_CSP = r"default-src 'none'; img-src 'self' data:; media-src 'self' data:; style-src 'self' data: 'unsafe-inline'; font-src 'self' data:; frame-src 'self' data:"
+# These are MIME types the `mimetypes` library doesn't recognize or gets wrong. Matches were found at:
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types
+MIME_TYPE_TO_FILE_EXTENSION = {
+    'application/font-woff': '.woff',
+    'application/font-woff2': '.woff2',
+    'application/vnd.ms-fontobject': '.eot',
+    'font/opentype': '.otf',
+    'font/ttf': '.ttf',
+    'font/woff2': '.woff2',
+    'image/jpeg': '.jpg',
+    'image/webp': '.webp',
+}
+
+
+@command()
+@option('--preserve-originals/--no-preserve-originals',
+        default=True,
+        help='Save original html files in a newly created `originals` directory in IN_DIRECTORY (default: True)')
+@argument('in_directory', type=Path(exists=True, file_okay=False))
+def main(in_directory, preserve_originals):
+    """
+    Extracts resources from the html pages in IN_DIRECTORY and stores them in a separate directory for Git-LFS storage.
+    """
+    if preserve_originals:
+        originals_dir = pathlib.Path(in_directory) / 'originals'
+        originals_dir.mkdir(parents=True, exist_ok=True)
+
+    for file in pathlib.Path(in_directory).iterdir():
+        if file.suffix != '.html':
+            print(f'Skipping {file.name}; not an html file')
+            continue
+        if preserve_originals:
+            shutil.copyfile(file, originals_dir / file.name)
```

Comment:
Also, it'd save IO to *move* the original and then re-write the new (smaller) thing in the old place.

Reply:
When you say:
> It would be spiffy if this didn't overwriting existing files.

do you mean you'd prefer a new file always be made or are you saying (along with the second comment here) that you'd rather this code path move the file instead of copy?

Reply:
Neither, actually! What I meant was, since, when you create the originals dir, you say it's okay if it already exists, it makes me nervous to go around overwriting the contents, since it could be that somebody manually made an originals dir and was sticking conflictingly-named things in it. What I would do is crash if there is an existing file there with a conflicting name. Or, if you're feeling fancy, start sticking numbers after the filename until it no longer conflicts (`313 2.html`, `313 3.html`, etc.). But it's not clear to me that the latter is even preferable, so I'd just let it crash, ideally with a nice message.

Reply:
I get you now. I also like saying it's not okay if the `originals` directory already exists. It feels the safest.

Reply:
Great. Easiest, too!